lxd/cluster/member_state.go

```diff
@@ -81,6 +86,66 @@ func LocalSysInfo() (*api.ClusterMemberSysInfo, error) {
 	return sysInfo, nil
 }
 
+// ClusterSysInfo returns a map from clusterMemberName -> sysinfo for every member
+// of the cluster. This requires an HTTP API call to the rest of the cluster.
+// Fails with ClusterUnavailableError if any requested member is offline.
+func ClusterSysInfo(s *state.State, members []db.NodeInfo) (map[string]api.ClusterMemberSysInfo, error) {
```

Comment:
Rather than duplicate the NewNotifer logic, the changes you've needed here sound quite useful to add to NewNotifier and the returned Notifier function definition anyway.

So I'm thinking we should update NewNotifier to:

```go
type Notifier func(member db.NodeInfo, hook func(lxd.InstanceServer) error) error

func NewNotifier(state *state.State, networkCert *shared.CertInfo, serverCert *shared.CertInfo, policy NotifierPolicy, members ...[]db.NodeInfo) (Notifier, error) {
```

This would make `members` optional, and if not supplied then all members would be loaded as they are today. 

Also there looks to be some optimisations we can make in NewNotifier that you've used here:

1. Avoid the call to tx.GetNodeOfflineThreshold(ctx) and s.GlobalConfig.OfflineThreshold() instead.
2. Change `if localClusterAddress == "" {` to `if s.ServerClustered {`.

That way we get to use the more resilient down detection in NewNotifier here too.

Reply:
We probably still need the ClusterSysInfo function for convenience, but it can use the updated NewNotifier under the hood.

Reply:
I had considered it but decided against it while drafting this since updating all the usages was going to take a while. Agree though, I'll put together a PR.